src/hooks/useFetchThemes.tsx

```diff
@@ -28,21 +28,26 @@ const useFetchThemes = (
 		const fetchData = async () => {
 			setLoading(true);
 			try {
-				let apiThemes;
+				let apiThemes = null;
 				if (url.startsWith("http")) {
 					let finalUrl = `${url}?pageSize=${pageSize}&pageNum=${pageNum}`;
 					if (searchQuery) {
 						finalUrl += `&searchQuery=${encodeURIComponent(searchQuery)}`;
 					}
 
 					const response = await galleryApiFetch(finalUrl);
-					apiThemes = await response.json();
+					const result = await response.json();
+          apiThemes = result.data;
 				} else {
 					apiThemes = Placeholders.themes;
 				}
 
-				const themes = await fetchThemesFromGitHub(apiThemes);
-				setThemes(themes);
+				if (apiThemes) {
```

Comment:
I might be missing something here but wouldn't apiThemes always be present/not null because it's either the API response or the placeholders?

could also initialize with let apiThemes = Placeholder.themes instead of null if that's the case too. or is there a case the response could be null?

Reply:
You're right that the response can actually be null - if `success` is `false` then the `data` field will be absent 😝

Reply:
gotcha. Looks good to me, if so.